app/models/upload_reference.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class UploadReference < ActiveRecord::Base
+  belongs_to :upload
+  belongs_to :target, polymorphic: true
+
+  def self.ensure_exist!(upload_ids: [], target: nil, target_type: nil, target_id: nil)
+    raise "target OR target_type and target_id are required" if !target && !(target_type && target_id)
+
+    if target.present?
+      target_type = target.class
+      target_id = target.id
+    end
+
+    upload_ids = upload_ids.uniq.reject(&:blank?)
```

Comment:
I think `.compact` will work here rather than `.reject(&:blank?)` (unless maybe you are expecting these to sometimes be `""`?

Reply:
I noticed that some invocations actually have IDs as strings and that is exactly why I made it this way. If I remember correctly, some tests fail with `.compact`.

Reply:
No problem!